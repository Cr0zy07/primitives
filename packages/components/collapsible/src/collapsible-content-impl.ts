import { computed, defineComponent, h, mergeProps, nextTick, reactive, ref, toRefs, watchEffect } from 'vue'
import { reactiveOmit, useComposedRefs, useForwardRef } from '@oku-ui/use-composable'
import { Primitive } from '@oku-ui/primitive'
import { getState } from './utils'
import { COLLAPSIBLE_CONTENT_IMPL_NAME, COLLAPSIBLE_CONTENT_NAME, collapsibleContentImplProps, scopeCollapsibleProps, useCollapsibleInject } from './props'
import type { CollapsibleContentImplElement, CollapsibleContentImplNativeElement } from './props'

const collapsibleContentImpl = defineComponent({
  name: COLLAPSIBLE_CONTENT_IMPL_NAME,
  inheritAttrs: false,
  props: {
    ...collapsibleContentImplProps.props,
    ...scopeCollapsibleProps,
  },
  emits: collapsibleContentImplProps.emits,
  setup(props, { attrs, slots }) {
    const { scopeOkuCollapsible, present, ...contentProps } = toRefs(props)

    const _reactive = reactive(contentProps)
    const otherProps = reactiveOmit(_reactive, (key, _value) => key === undefined)

    const forwardedRef = useForwardRef()

    const inject = useCollapsibleInject(COLLAPSIBLE_CONTENT_NAME, scopeOkuCollapsible.value)

    const isPresent = ref(present.value)
    const collapsibleContentRef = ref<CollapsibleContentImplElement | null>(null)
    const composedRefs = useComposedRefs(forwardedRef, collapsibleContentRef)
    const heightRef = ref<number | undefined>(0)
    const height = computed(() => heightRef.value)
    const widthRef = ref<number | undefined>(0)
    const width = computed(() => widthRef.value)
    // when opening we want it to immediately open to retrieve dimensions
    // when closing we delay `present` to retrieve dimensions before closing
    const isOpen = computed(() => inject.open.value || isPresent.value)
    const isMountAnimationPreventedRef = ref(isOpen.value)
    const originalStylesRef = ref<Record<string, string>>()

    // const rAf = ref()

    // onMounted(() => {
    //   rAf.value = requestAnimationFrame(() => (isMountAnimationPreventedRef.value = false))
    // })

    // onBeforeUnmount(() => {
    //   cancelAnimationFrame(rAf.value)
    // })

    watchEffect((onInvalidate) => {
      const rAf = requestAnimationFrame(() => (isMountAnimationPreventedRef.value = false))

      onInvalidate(() => cancelAnimationFrame(rAf))
    })

    watchEffect(async () => {
      const node = collapsibleContentRef.value
      if (node) {
        originalStylesRef.value = originalStylesRef.value || {
          transitionDuration: node.style.transitionDuration,
          animationName: node.style.animationName,
        }
        // block any animations/transitions so the element renders at its full dimensions
        node.style.transitionDuration = '0s'
        node.style.animationName = 'none'

        // get width and height from full dimensions
        const rect = node.getBoundingClientRect()
        heightRef.value = rect.height
        widthRef.value = rect.width

        await nextTick()

        // kick off any animations/transitions that were originally set up if it isn't the initial mount
        if (!isMountAnimationPreventedRef.value) {
          node.style.transitionDuration = originalStylesRef.value.transitionDuration
          node.style.animationName = originalStylesRef.value.animationName
        }

        isPresent.value = present.value
      }
    })

    return () => h(Primitive.div, {
      'data-state': getState(inject.open.value),
      'data-disabled': inject.disabled?.value ? '' : undefined,
      'id': inject.contentId.value,
      'hidden': !isOpen.value,
      ...mergeProps(attrs, otherProps),
      'ref': composedRefs,
      'style': {
        ['--oku-collapsible-content-height' as any]: height.value ? `${height.value}px` : undefined,
        ['--oku-collapsible-content-width' as any]: width.value ? `${width.value}px` : undefined,
        ...attrs.style as any,
      },
    }, () => isOpen.value && slots.default?.())
  },
})

// TODO: https://github.com/vuejs/core/pull/7444 after delete
export const OkuCollapsibleContentImpl = collapsibleContentImpl as typeof collapsibleContentImpl & (new () => { $props: CollapsibleContentImplNativeElement })
