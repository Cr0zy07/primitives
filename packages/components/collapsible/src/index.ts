export { OkuCollapsible } from './collapsible'
export { OkuCollapsibleTrigger } from './collapsible-trigger'
export { OkuCollapsibleContent } from './collapsible-content'
export { OkuCollapsibleContentImpl } from './collapsible-content-impl'

export type {
  CollapsibleProps,
  CollapsibleEmits,
  CollapsibleElement,
  CollapsibleNativeElement,
} from './props'

export type {
  CollapsibleContentProps,
  CollapsibleContentElement,
  CollapsibleContentNativeElement,
} from './props'

export type {
  CollapsibleContentImplProps,
  CollapsibleContentImplElement,
  CollapsibleContentImplNativeElement,
} from './props'

export type {
  CollapsibleTriggerProps,
  CollapsibleTriggerEmits,
  CollapsibleTriggerElement,
  CollapsibleTriggerNativeElement,
} from './props'

export {
  collapsibleContentProps,
  collapsibleProps,
  collapsibleTriggerProps,
  createCollapsibleScope,
} from './props'
