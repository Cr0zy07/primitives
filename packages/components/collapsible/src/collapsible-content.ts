import type { ComputedRef } from 'vue'
import { computed, defineComponent, h, mergeProps, reactive, toRefs } from 'vue'
import { reactiveOmit, useForwardRef } from '@oku-ui/use-composable'
import { OkuPresence } from '@oku-ui/presence'
import { OkuCollapsibleContentImpl } from './collapsible-content-impl'
import { COLLAPSIBLE_CONTENT_NAME, collapsibleContentProps, scopeCollapsibleProps, useCollapsibleInject } from './props'
import type { CollapsibleContentNativeElement } from './props'

const collapsibleContent = defineComponent({
  name: COLLAPSIBLE_CONTENT_NAME,
  components: {
    OkuCollapsibleContentImpl,
  },
  inheritAttrs: false,
  props: {
    ...collapsibleContentProps.props,
    ...scopeCollapsibleProps,
  },
  emits: collapsibleContentProps.emits,
  setup(props, { attrs, slots }) {
    const { forceMount, ...contentProps } = toRefs(props)

    const _reactive = reactive(contentProps)
    const otherProps = reactiveOmit(_reactive, (key, _value) => key === undefined)

    const forwardedRef = useForwardRef()

    const inject = useCollapsibleInject(COLLAPSIBLE_CONTENT_NAME, props.scopeOkuCollapsible)

    return () => h(OkuPresence, {
      present: computed(() => forceMount.value || inject.open.value).value,
    }, {
      default: ({ isPresent }: { isPresent: ComputedRef<boolean> }) => h(OkuCollapsibleContentImpl, {
        ...mergeProps(attrs, otherProps),
        ref: forwardedRef,
        present: isPresent.value,
      }, () => slots.default?.()),
    })
  },
})

// TODO: https://github.com/vuejs/core/pull/7444 after delete
export const OkuCollapsibleContent = collapsibleContent as typeof collapsibleContent & (new () => { $props: CollapsibleContentNativeElement })
