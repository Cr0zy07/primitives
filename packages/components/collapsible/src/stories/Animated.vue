<script setup lang="ts">
import { OkuCollapsible, OkuCollapsibleContent, OkuCollapsibleTrigger } from '@oku-ui/collapsible'
</script>

<template>
  <h1>Closed by default</h1>
  <OkuCollapsible class="collapsible">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-animated-content">
      <div :style="{ padding: '10px' }">
        Content 1
      </div>
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h1>Open by default</h1>
  <OkuCollapsible default-open class="collapsible">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-animated-content">
      <div :style="{ padding: '10px' }">
        Content 1
      </div>
    </OkuCollapsibleContent>
  </OkuCollapsible>
</template>
