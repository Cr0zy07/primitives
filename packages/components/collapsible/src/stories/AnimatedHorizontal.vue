<script setup lang="ts">
import { OkuCollapsible, OkuCollapsibleContent, OkuCollapsibleTrigger } from '@oku-ui/collapsible'
</script>

<template>
  <OkuCollapsible class="collapsible">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-animated-width-content">
      <div :style="{ padding: '10px' }">
        Content
      </div>
    </OkuCollapsibleContent>
  </OkuCollapsible>
</template>
