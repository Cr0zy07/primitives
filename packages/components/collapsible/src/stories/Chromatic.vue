<script setup lang="ts">
import { OkuCollapsible, OkuCollapsibleContent, OkuCollapsibleTrigger } from '@oku-ui/collapsible'
</script>

<template>
  <h1>Uncontrolled</h1>
  <h2>Closed</h2>
  <OkuCollapsible class="collapsible">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h2>Open</h2>
  <OkuCollapsible class="collapsible" default-open>
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h1>Controlled</h1>
  <h2>Closed</h2>
  <OkuCollapsible class="collapsible" :open="false">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h2>Open</h2>
  <OkuCollapsible class="collapsible" open>
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h1>Disabled</h1>
  <OkuCollapsible class="collapsible" disabled>
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h1>State attributes</h1>
  <h2>Closed</h2>
  <OkuCollapsible class="collapsible-attr-styles">
    <OkuCollapsibleTrigger class="collapsible-attr-styles">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent
      class="collapsible-attr-styles"
      :style="{
        // ensure we can see the content (because it has `hidden` attribute)
        display: 'block',
      }"
    >
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h2>Open</h2>
  <OkuCollapsible class="collapsible-attr-styles" default-open>
    <OkuCollapsibleTrigger class="collapsible-attr-styles">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent
      class="collapsible-attr-styles"
      :style="{
        // ensure we can see the content (because it has `hidden` attribute)
        display: 'block',
      }"
    >
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h2>Disabled</h2>
  <OkuCollapsible class="collapsible-attr-styles" default-open disabled>
    <OkuCollapsibleTrigger class="collapsible-attr-styles">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent
      class="collapsible-attr-styles"
      :style="{
        // ensure we can see the content (because it has `hidden` attribute)
        display: 'block',
      }"
    >
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>
</template>
