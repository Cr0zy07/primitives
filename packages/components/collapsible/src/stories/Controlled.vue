<script setup lang="ts">
import { ref } from 'vue'
import { OkuCollapsible, OkuCollapsibleContent, OkuCollapsibleTrigger } from '@oku-ui/collapsible'

const open = ref(false)
</script>

<template>
  <OkuCollapsible :open="open" class="collapsible" @open-change="open = $event">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      {{ open ? 'close' : 'open' }}
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content" as-child>
      <article>Content 1</article>
    </OkuCollapsibleContent>
  </OkuCollapsible>
</template>
