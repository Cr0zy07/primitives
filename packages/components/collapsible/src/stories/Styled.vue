<script setup lang="ts">
import { OkuCollapsible, OkuCollapsibleContent, OkuCollapsibleTrigger } from '@oku-ui/collapsible'
</script>

<template>
  <OkuCollapsible class="collapsible">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>
</template>
