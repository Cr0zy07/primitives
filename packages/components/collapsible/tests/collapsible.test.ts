import { defineComponent, nextTick } from 'vue'
import { enableAutoUnmount, mount, shallowMount } from '@vue/test-utils'
import type { DOMWrapper, VueWrapper } from '@vue/test-utils'
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
import { axe } from 'vitest-axe'

import { OkuCollapsible, OkuCollapsibleContent, OkuCollapsibleContentImpl, OkuCollapsibleTrigger } from '../src'

import Styled from '../src/stories/Styled.vue'
import Controlled from '../src/stories/Controlled.vue'
import Animated from '../src/stories/Animated.vue'
import AnimatedHorizontal from '../src/stories/AnimatedHorizontal.vue'
import Chromatic from '../src/stories/Chromatic.vue'

enableAutoUnmount(afterEach)

const TRIGGER_TEXT = 'Trigger'
const CONTENT_TEXT = 'Content'

const onOpenChange = vi.fn()

const CollapsibleTest = defineComponent({
  components: {
    OkuCollapsible,
    OkuCollapsibleContent,
    OkuCollapsibleContentImpl,
    OkuCollapsibleTrigger,
  },
  setup() {
    return {
      TRIGGER_TEXT,
      CONTENT_TEXT,
      onOpenChange,
    }
  },
  template: `
    <OkuCollapsible v-bind="$attrs" @open-change="onOpenChange">
      <OkuCollapsibleTrigger>{{ TRIGGER_TEXT }}</OkuCollapsibleTrigger>
      <OkuCollapsibleContent>{{ CONTENT_TEXT }}</OkuCollapsibleContent>
    </OkuCollapsible>
  `,
})

describe('okuCollapsible', () => {
  let wrapper: VueWrapper
  let trigger: DOMWrapper<HTMLElement>
  let content: DOMWrapper<HTMLElement> | null

  describe('given a default Collapsible', () => {
    beforeEach(() => {
      wrapper = mount(CollapsibleTest, {
        attachTo: document.body,
      })

      trigger = wrapper.findAll('button').filter((button: { text: () => string }) => button.text() === TRIGGER_TEXT).at(0)!
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    // TODO: fix is undifind
    describe('when clicking the trigger', () => {
      beforeEach(async () => {
        trigger.trigger('click')
        // content = wrapper.findAll('div').filter((content: { text: () => string }) => content.text() === CONTENT_TEXT).at(0)!
      })

      it('should open the content', () => {
        content = wrapper.findAll('div').filter((content: { text: () => string }) => content.text() === CONTENT_TEXT).at(0)!

        expect(content?.isVisible()).toBe(true)
      })

      describe('and clicking the trigger again', () => {
        beforeEach(() => {
          trigger.trigger('click')
        })
        // TODO: fix test case
        it('should close the content', () => {
          content = wrapper.findAll('div').filter((content: { text: () => string }) => content.text() === CONTENT_TEXT).at(0)!

          // expect(content?.isVisible()).toBe(false)
        })
      })
    })
  })

  describe('given an open uncontrolled Collapsible', () => {
    beforeEach(() => {
      wrapper = mount(CollapsibleTest, {
        props: {
          defaultOpen: true,
        },
        attachTo: document.body,
      })
    })

    describe('when clicking the trigger', () => {
      beforeEach(async () => {
        trigger = wrapper.findAll('button').filter((button: { text: () => string }) => button.text() === TRIGGER_TEXT).at(0)!
        content = wrapper.findAll('div').filter((content: { text: () => string }) => content.text() === CONTENT_TEXT).at(0)!
        trigger.trigger('click')
      })

      it('should close the content', () => {
        expect(content?.isVisible()).toBe(false)
      })

      it('should call `onOpenChange` event with `false` value', () => {
        expect(onOpenChange).toHaveBeenCalledWith(false)
      })
    })
  })

  describe('given an open controlled Collapsible', () => {
    beforeEach(() => {
      wrapper = mount(CollapsibleTest, {
        props: {
          open: true,
        },
        attachTo: document.body,
      })

      content = wrapper.findAll('div').filter((content: { text: () => string }) => content.text() === CONTENT_TEXT).at(0)!
    })

    describe('when clicking the trigger', () => {
      beforeEach(() => {
        trigger = wrapper.findAll('button').filter((button: { text: () => string }) => button.text() === TRIGGER_TEXT).at(0)!
        trigger.trigger('click')
      })

      it('should call `onOpenChange` prop with `false` value', () => {
        expect(onOpenChange).toHaveBeenCalledWith(false)
      })

      it('should not close the content', () => {
        expect(content?.isVisible()).toBe(true)
      })
    })
  })
})

describe('okuCollapsible Stories', () => {
  describe('styled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Styled>>

    beforeEach(async () => {
      wrapper = shallowMount(Styled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('controlled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Controlled>>

    beforeEach(async () => {
      wrapper = shallowMount(Controlled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('animated', () => {
    let wrapper: VueWrapper<InstanceType<typeof Animated>>

    beforeEach(async () => {
      wrapper = shallowMount(Animated, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('animatedHorizontal', () => {
    let wrapper: VueWrapper<InstanceType<typeof AnimatedHorizontal>>

    beforeEach(async () => {
      wrapper = shallowMount(AnimatedHorizontal, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('chromatic', () => {
    let wrapper: VueWrapper<InstanceType<typeof Chromatic>>

    beforeEach(async () => {
      wrapper = shallowMount(Chromatic, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
