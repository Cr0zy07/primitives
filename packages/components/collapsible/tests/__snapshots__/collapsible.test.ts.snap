// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuCollapsible Stories > animated > should render correctly 1`] = `
"<h1>Closed by default</h1>
<oku-collapsible-stub disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>
<h1>Open by default</h1>
<oku-collapsible-stub defaultopen=\\"true\\" disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>"
`;

exports[`okuCollapsible Stories > animatedHorizontal > should render correctly 1`] = `"<oku-collapsible-stub disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>"`;

exports[`okuCollapsible Stories > chromatic > should render correctly 1`] = `
"<h1>Uncontrolled</h1>
<h2>Closed</h2>
<oku-collapsible-stub disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>
<h2>Open</h2>
<oku-collapsible-stub defaultopen=\\"true\\" disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>
<h1>Controlled</h1>
<h2>Closed</h2>
<oku-collapsible-stub open=\\"false\\" disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>
<h2>Open</h2>
<oku-collapsible-stub open=\\"true\\" disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>
<h1>Disabled</h1>
<oku-collapsible-stub disabled=\\"true\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>
<h1>State attributes</h1>
<h2>Closed</h2>
<oku-collapsible-stub disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible-attr-styles\\"></oku-collapsible-stub>
<h2>Open</h2>
<oku-collapsible-stub defaultopen=\\"true\\" disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible-attr-styles\\"></oku-collapsible-stub>
<h2>Disabled</h2>
<oku-collapsible-stub defaultopen=\\"true\\" disabled=\\"true\\" aschild=\\"false\\" class=\\"collapsible-attr-styles\\"></oku-collapsible-stub>"
`;

exports[`okuCollapsible Stories > controlled > should render correctly 1`] = `"<oku-collapsible-stub open=\\"false\\" disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>"`;

exports[`okuCollapsible Stories > styled > should render correctly 1`] = `"<oku-collapsible-stub disabled=\\"false\\" aschild=\\"false\\" class=\\"collapsible\\"></oku-collapsible-stub>"`;
